### PR TITLE
Support include/exclude containers in ECS Fargate (process-agent)

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Add support to include/exclude containers in ECS Fargate (live container view, process-agent).
Fix support of `container_include_metrics/container_exclude_metrics` in container checks.

### Describe your test plan

Start an ECS cluster with `DD_CONTAINER_EXCLUDE_METRICS = image:fg-proxy name:.*dummy.*` (for instance) and verify that the other containers in your task are properly excluded in:
- Live container view
- Metrics `ecs.fargate.cpu.user`

### CODE REVIEW

File: pkg/util/containers/filter.go [hunk 1]
```diff
@@ -38,13 +38,13 @@ const (
...
-	ImageBlacklist     []*regexp.Regexp
-	NameBlacklist      []*regexp.Regexp
-	NamespaceBlacklist []*regexp.Regexp
+	Enabled              bool
+	ImageIncludeList     []*regexp.Regexp
+	NameIncludeList      []*regexp.Regexp
+	NamespaceIncludeList []*regexp.Regexp
+	ImageExcludeList     []*regexp.Regexp
+	NameExcludeList      []*regexp.Regexp
+	NamespaceExcludeList []*regexp.Regexp
```
Comment (sunhay) [inline pkg/util/containers/filter.go:47]: 🙌  🙇

Comment (ahmed-mez) [reply pkg/util/containers/filter.go:47]: 👏 🖤

File: pkg/util/containers/filter.go [hunk 2]
```diff
@@ -99,46 +99,49 @@ func ResetSharedFilter() {
...
 
 // NewFilter creates a new container filter from a two slices of
-// regexp patterns for a whitelist and blacklist. Each pattern should have
+// regexp patterns for a include list and exclude list. Each pattern should have
 // the following format: "field:pattern" where field can be: [image, name].
 // An error is returned if any of the expression don't compile.
-func NewFilter(whitelist, blacklist []string) (*Filter, error) {
-	iwl, nwl, nswl, err := parseFilters(whitelist)
+func NewFilter(includeList, excludeList []string) (*Filter, error) {
+	iwl, nwl, nswl, err := parseFilters(includeList)
```
Comment (ahmed-mez) [inline pkg/util/containers/filter.go:106]: nit: here `wl` in the variable names stands for `whitelist`, let's rename them

File: pkg/util/containers/filter.go [hunk 3]
```diff
@@ -99,46 +99,49 @@ func ResetSharedFilter() {
...
 	if err != nil {
 		return nil, err
 	}
-	ibl, nbl, nsbl, err := parseFilters(blacklist)
+	ibl, nbl, nsbl, err := parseFilters(excludeList)
```
Comment (ahmed-mez) [inline pkg/util/containers/filter.go:110]: nit: here `bl` in the variable names stands for `blacklist`, let's rename them

File: pkg/util/containers/filter.go [hunk 4]
```diff
@@ -99,46 +99,49 @@ func ResetSharedFilter() {
...
 }
 
-// NewFilterFromConfig creates a new container filter, sourcing patterns
-// from the pkg/config options
-func NewFilterFromConfig() (*Filter, error) {
-	whitelist := config.Datadog.GetStringSlice("container_include")
-	blacklist := config.Datadog.GetStringSlice("container_exclude")
-	if len(whitelist) == 0 {
+// newMetricFilterFromConfig creates a new container filter, sourcing patterns
+// from the pkg/config options, to be used only for metrics
```
Comment (ahmed-mez) [inline pkg/util/containers/filter.go:127]: could you please document why we merge `container_exclude/include` and `container_exclude/include_metrics` here for future reference?

File: pkg/util/ecs/common.go [hunk 5]
```diff
@@ -36,8 +36,17 @@ func ListContainersInCurrentTask() ([]*containers.Container, error) {
 		log.Error("Unable to get the container list from ecs")
 		return cList, err
 	}
+
+	filter, err := containers.GetSharedMetricFilter()
+	if err != nil {
+		log.Warnf("Unable to get container filter. All containers in ECS Task will be processed, err: %v", err)
```
Comment (ahmed-mez) [inline pkg/util/ecs/common.go:42]: `ListContainersInCurrentTask` is called continuously so this warning might become spammy - let's switch to debug wdyt?

**Review by ahmed-mez — APPROVED** [review]
Few nits, looks great otherwise 👍

File: pkg/util/ecs/common.go (hunk 5 above)
Comment (vboulineau) [reply pkg/util/ecs/common.go:42]: I believe it should be spammy, customers would be paying for containers they don't want.

File: pkg/util/containers/filter.go (hunk 4 above)
Comment (vboulineau) [reply pkg/util/containers/filter.go:127]: Fixed

**Pull request merged**